### ISSUE TITLE
Numeric masks not working as intended

### ISSUE THREAD

**Issue by fayne1337** [filamentphp/filament#3049]
### Package

filament/forms

### Package Version

v2.13.9

### Laravel Version

v9.17.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.6

### Bug description

Masked numeric fields seem not to behave as intended via https://imask.js.org/guide.html#masked-number

Testing in a custom file as per the documentation, all works fine. Upon entering a number and then tabbing out, the value is being formatted correctly.

```
IMask(document.getElementById('test'), {
    mask: Number,
    padFractionalZeros: true,
});
```

![working](https://user-images.githubusercontent.com/108292004/177606107-0af6ec51-5f3f-4387-8119-01505ae88c75.gif)

When trying to replicate this behavior in a filament `TextInput` the field breaks. Immediately after typing a single digit, the field is being formatted. It is then not possible to move into the decimals by pressing comma/dot/tab but by either using the right arrow key or the mouse. Even if I do enter the decimals, it is not possible to override any of them. 

```
Components\TextInput::make('amount')
    ->mask(
        fn(Mask $mask) => $mask
            ->numeric()
            ->padFractionalZeros()
    ),
```
![not_working](https://user-images.githubusercontent.com/108292004/177606950-1dd76732-9c13-48c4-99cf-cbd8845452ba.gif)

**Comment by zepfietje** [comment]
Can you provide us a reproduction repository URL and then we'll look at your issue?

That would allow us to download it and review your bug much easier, so it can be fixed quicker. Please make sure to give us some instructions on where to find the bug, and include a database seeder with everything we need to set it up quickly.

**Comment by danharrin** [comment]
Fixed by #3441.
